File: Golang/7. Reverse Integer.go
package leetcode

// 整数反转
// https://leetcode-cn.com/problems/reverse-integer
func reverse7(x int) int {
	var ans int
	for x != 0 {
		ans = ans*10 + x%10
		x /= 10
	}
	// 范围在[−2^31, 2^31−1]
	if ans > 1<<31-1 || ans < -(1<<31) {
		return 0
	}
	return ans
}

File: Python/7_reverse_integer.py
# -*- coding: utf-8 -*-
# python 3
# 需要考虑溢出

# Description
#
# Reverse digits of an integer.


class Solution(object):
    def reverse(self, n):
        """
        :type n: int
        :rtype: int
        """
        x = 0
        y = n if n > 0 else -n
        while y:
            x, y = x * 10 + y % 10, y // 10
        if x > 2 ** 31 or x < -(2 ** 31):
            return 0
        return x if n > 0 else -x


if __name__ == '__main__':
    print Solution().reverse(123)
    print Solution().reverse(-123)

